portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java

```diff
@@ -1738,13 +1738,33 @@ public void includeBottom(PortalRenderContext rcontext, Site site)
                         			log.error(e1.getMessage(), e1);
                         		}
                         	}
-						}
+                        }
 
                         if(sakaiThemeSwitcherEnabled) {
                             rcontext.put("themeSwitcher", true);
+
+                            boolean darkTheme = false;
+                            int themeUserPref = -1;
+                            
+                            ResourceProperties themePrefs = prefs.getProperties(org.sakaiproject.user.api.PreferencesService.USER_SELECTED_UI_THEME_PREFS);
+                            try {
+                                    darkTheme = themePrefs.getBooleanProperty("darkTheme");
+                                    
+                                    //temporary:
+                                    if(themePrefs.getBooleanProperty("darkTheme")) {
+                                        themeUserPref = 1;
+                                    } else {
+                                        themeUserPref = 0;
+                                    }
+                            }
+                            catch (Exception any) { 
+                                    // We'll default to light theme
+                            }
+                            
+                            rcontext.put("darkTheme", Boolean.valueOf(darkTheme));
```

Comment:
`Boolean.valueOf()` not necessary here:

```suggestion
                            rcontext.put("darkTheme", darkTheme);
```